### ISSUE TITLE
Clipboard handling of contact address often useless

### ISSUE THREAD

**Issue by neocogent** [spesmilo/electrum#1734]
When you copy a contact address to the clipboard it copies the name and &lt;address&gt; . But when you paste it pastes just the name and not the address - which is 100% useless, of course. In particular I want to copy several contact addresses into a "Pay To Many" send trx. This is not possible with the current clipboard handling. In fact, it would just be better if copy/paste of addresses just stuck to address only as many uses cannot correctly decode a combination "Name &lt;address&gt;" pair, and as far as I know only the "single payee send trx" seems to work with this, a very limited use of the copy/paste function.

From the contact address list - here is how I have to make a Pay to many trx. I have to select "Pay to" because "Copy to Clipboard" is stupid. Then I double click on the send trx address and copy/paste to a text document. I repeat this with several contacts to make up a list in a text document. Then I can add amounts and copy/paste into a "pay to many" trx.Yikes, talk about clumsy user interface.

**Comment by neocogent** [comment]
I've noticed now that when you paste a single-word-name contact it actually pastes and prompts for completion with the address. But when you have a contact name that includes a space then it does not. So maybe the deviation from expectations here is that it cannot properly handle spaces in contact names.

But also, when copy/pasting outside the program it really does need to paste address not contact name.

**Comment by fortran77** [comment]
Many applications interpret a double-click in some intelligent manner designed to make life easier for users.

Much of the time this works reasonably well. But some of the time it just frustrates the user.

In Electrum, for example, if we're looking at Addresses, and we double-click on the Balance field wanting to select and copy it, we get a context menu that that lets us copy the address but not the balance. Only after three menus deep do we finally get a chance to select and copy the balance.

If we're looking at the history, and we double-click on the Amount or Balance field, we get a context menu off which we get other menus, from which we can select and copy any of many other fields -- _except_ the Amount or Balance field.

Copying to the clipboard should always work like this: Select or copy the field that the user is pointing to.

Not only will this let users choose what to copy, but also, they won't have to learn the specific workings of the copy operation for each application.

This does raise the question of how to allow copying if we are using double-click to show a context menu instead of selecting text. Solution: Include the corresponding copy in the context menu instead. Then the user is never more than a click away from copying the field on which he double-clicked.

**Comment by ecdsa** [comment]
@fortran77 so, i I understand you well, the context menu should adapt to the selected row

**Comment by fortran77** [comment]
Not the column, but rather, the cell (which I called the "field").

Data presented in a format that looks like a table should also act like a table. A user double-clicking on a specific cell of that table probably wants to do something with that specific cell, and we should let him.

**Comment by ecdsa** [comment]
ok, let me know if what I pushed suits your needs.

**Comment by fortran77** [comment]
I see that you have implemented exactly what I proposed. On the History, Addresses, and Contacts screens, a right-click on any field gave me a context menu allowing the text in that field to be copied to the clipboard.

(I found only one exception, which probably doesn't matter. On the Addresses screen, when an address is preceded by a +, and I click on the + to get an indented line below it containing a long transaction ID and an optional label, a right-click on those fields doesn't result in any context menu.)

This is very gratifying, and also a bit scary. It's gratifying that my editorial opinions about user interface design turned into so much code so quickly. It's scary for the same reason.
